server/command/command.go

```diff
@@ -0,0 +1,284 @@
+package command
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/bakurits/mattermost-plugin-splunk/server/api"
+	"github.com/bakurits/mattermost-plugin-splunk/server/config"
+	"github.com/bakurits/mattermost-plugin-splunk/server/splunk"
+
+	"github.com/google/uuid"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const (
+	helpTextHeader = "###### Mattermost Splunk Plugin - Slash command help\n"
+	helpText       = `
+* |/splunk help| - print this help message
+* |/splunk auth --login [server base url] [username] [password]| - log into the splunk server
+* |/splunk alert --subscribe| - subscribe to alerts
+* |/splunk logs --list| - list names of logs on server
+* |/splunk log [logname]| - show specific log from server
+`
+	autoCompleteDescription = ""
+	autoCompleteHint        = ""
+	pluginDescription       = ""
+	slashCommandName        = "splunk"
+)
+
+// Handler returns API for interacting with plugin commands
+type Handler interface {
+	Handle(args ...string) (*model.CommandResponse, error)
+}
+
+// HandlerFunc command handler function type
+type HandlerFunc func(args ...string) (*model.CommandResponse, error)
+
+// HandlerMap map of command handler functions
+type HandlerMap struct {
+	handlers       map[string]HandlerFunc
+	defaultHandler HandlerFunc
+}
+
+// NewHandler returns new Handler with given dependencies
+func NewHandler(args *model.CommandArgs, conf *config.Config, a splunk.Splunk) Handler {
+	return newCommand(args, conf, a)
+}
+
+// GetSlashCommand returns command to register
+func GetSlashCommand() *model.Command {
+	return &model.Command{
+		Trigger:          slashCommandName,
+		DisplayName:      slashCommandName,
+		Description:      pluginDescription,
+		AutoComplete:     true,
+		AutoCompleteDesc: autoCompleteDescription,
+		AutoCompleteHint: autoCompleteHint,
```

Comment:
I recommend implementing slash command autocomplete by supplying the `AutocompleteData` field. Here's an example from the Jira plugin https://github.com/mattermost/mattermost-plugin-jira/blob/59efc0f7d21785a109ba80b18943cdd3110fec11/server/command.go#L136

Reply:
https://github.com/bakurits/mattermost-plugin-splunk/issues/29